subsys/bluetooth/host/classic/ssp.c

```diff
@@ -443,6 +443,24 @@ void bt_hci_link_key_notify(struct net_buf *buf)
 
 	LOG_DBG("%s, link type 0x%02x", bt_addr_str(&evt->bdaddr), evt->key_type);
 
+	if (IS_ENABLED(CONFIG_BT_SMP_SC_ONLY) &&
+	    (!(evt->key_type == BT_LK_AUTH_COMBINATION_P256))) {
```

Comment:
`evt->key_type != BT_LK_AUTH_COMBINATION_P256` would be simpler (sorry if I missed it on the first review round)

Reply:
Yes. Updated.